src/Diagnostics.RuntimeHost/Controllers/DeploymentController.cs

```diff
@@ -0,0 +1,162 @@
+                }
+
+                // Otherwise, compile the detector to generate dll.
+                queryRes.CompilationOutput = await _compilerHostClient.GetCompilationResponse(file.Content, diagEntity.EntityType, references);  
+                
+                //  If compilation success, save dll to storage container.
+                if (queryRes.CompilationOutput.CompilationSucceeded)
+                {                                     
+                    var blobName = $"{detectorId.ToLower()}/{detectorId.ToLower()}.dll";
+                    DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"Deployment id {response.DeploymentGuid}, saving {blobName} to storage container");
+                    //  Save blob to storage account
+                    var etag = await storageService.LoadBlobToContainer(blobName, queryRes.CompilationOutput.AssemblyBytes);                
+                    if (string.IsNullOrWhiteSpace(etag))
+                    {
+                        throw new Exception($"Could not save changes {blobName} to storage");
+                    }
+                    response.DeployedDetectors.Add(detectorId);
+
+                    // Save entity to table
+                    diagEntity.Metadata = file.Metadata;
+                    diagEntity.GitHubSha = file.CommitId;
+                    byte[] asmData = Convert.FromBase64String(queryRes.CompilationOutput.AssemblyBytes);
+                    byte[] pdbData = Convert.FromBase64String(queryRes.CompilationOutput.PdbBytes);
+                    diagEntity = DiagEntityHelper.PrepareEntityForLoad(asmData, file.Content, diagEntity);
+
+                    DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"Deployment id {response.DeploymentGuid}, saving {diagEntity.RowKey} to storage table");
+                    await storageService.LoadDataToTable(diagEntity);
+
+                    DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"Deployment id {response.DeploymentGuid}, updating invoker cache for {diagEntity.RowKey}");
+                    // update invoker cache for detector. For gists, we dont need to update invoker cache as we pull latest code each time.
+                    if(diagEntity.PartitionKey.Equals("Detector"))
+                    {
+                        Assembly tempAsm = Assembly.Load(asmData, pdbData);
+                        EntityType entityType = EntityType.Detector;
+                        EntityMetadata metaData = new EntityMetadata(file.Content, entityType, null);
+                        var newInvoker = new EntityInvoker(metaData);
+                        newInvoker.InitializeEntryPoint(tempAsm);
+                        detectorCache.AddOrUpdate(detectorId, newInvoker);
+                    }
+                   
+                } else
+                {
+                    DiagnosticsETWProvider.Instance.LogRuntimeHostMessage($"Deployment id {response.DeploymentGuid}, compilation failed for {detectorId}");
```

Comment:
Once you add a new event for this, can we also add the reason for failure (i.e. compilation output)
It might help us in future to debug in case  partners wonder why their detector didnt deployed successfully.